bin/matrix-register.js

```diff
@@ -86,58 +86,68 @@ Matrix.localization.init(Matrix.localesFolder, Matrix.config.locale, function ()
                 description: result.description,
               };
 
-              // fire off worker
-              Matrix.firebase.device.add(deviceObj, events)
+              var duplicateDevices = _.values(Matrix.config.deviceMap).filter(function(d) {
+                return d.name === result.name;
+              });
 
-              // wrap this up
-              Matrix.firebase.user.watchForDeviceAdd(function(d){
-                var deviceId = d.key;
+              if(duplicateDevices.length === 0) {
```

Comment:
please catch and resolve the error condition first, that lines up more with node conventions.